runtime/interp.c

```diff
@@ -270,8 +270,7 @@ value caml_interprete(code_t prog, asize_t prog_size)
       (char *) raise_unhandled_code + sizeof(raise_unhandled_code),
       DIGEST_IGNORE, NULL);
 #ifdef THREADED_CODE
-    caml_instr_table = (char **) jumptable;
-    caml_instr_base = Jumptbl_base;
+    caml_init_thread_code(jumptable, Jumptbl_base);
```

Comment:
I wondered why the initialization function takes `void*` pointers instead of `char*` pointers. `jumptable` is defined as `void*`, but `Jumptbl_base` is currently defined as `char*` (so it is cast back and forth). According to the Gcc documentation, label-as-values really have type `void*`. In `fix_code.c` you use `char *` to be able to do pointer arithmetic.

I think the two following approaches would be slightly more consistent than the current one:
- keep everything as `void *` in interp.c, and `char *` in fix_code.c, which a cast in `caml_init_thread_code` (so: drop the `char *` casts in the Jumptbl_base definition)
- keep everything as `void *` everywhere, and cast right before doing the pointer substraction (so: same as above, plus changes to `fix_code.c`)